### DISCUSSION TITLE
Automated File and Folder Maintenance

### DISCUSSION THREAD

**Discussion by vincolay** [photoprism/photoprism#3984] [Ideas]
First off - THANK YOU for PhotoPrism. It's amazing and so appreciated.

Sorry if this is a duplicated feature idea but I didn't seem to find it in my search.

I've just moved onto a new server and outside of writing a script, can't find an easy way to:

1. Ingest media types and sources;
2. Rename all media files using exif data (year - month - date - number)
3. Move to a structured folder system based on media (Pictures > Year > Month)

Seems a lot of folks have this need but no one program can help and seems like a logical place to include this is PhotoPrism.

**Comment by lastzero** [comment]
Thank you very much! Automatic reorganization of originals could generally be implemented, however, there is the (big) risk that you rename all your files and the result is not what you wanted. Also, there are other tools that can help with this, see:

- https://docs.photoprism.app/getting-started/faq/#can-i-use-photoprism-to-sort-files-into-a-configurable-folder-structure

That's why we decided to tackle this later, when other high priority tasks are done and we don't need to rush it.

Note that you can already let PhotoPrism automatically organize new files if you use the import functionality and don't want to configure the file/directory names (this is not yet possible for the reasons mentioned above, e.g. because you could easily create naming conflicts with a custom scheme):

- https://docs.photoprism.app/user-guide/library/import/

**Reply by vincolay** [reply]
Thanks for your reply and thanks for all that you do!

**Reply by Moosbueffel** [reply]
A very helpfull and flexible tool I use ist:
https://toolsandmore.de/Central/Produkte/Software/Datei-Tools/Joe/

Alternatively you can use:
https://www.faststone.org/FSViewerDetail.htm
which is not only helpfull for renaming.
